data/kernels/blendop.cl

```diff
@@ -373,31 +373,34 @@ blendop_Lab (__read_only image2d_t in_a, __read_only image2d_t in_b, __read_only
   a /= scale;
   b /= scale;
 
-  const float4 min = (float4)(0.0f, -1.0f, -1.0f, 0.0f);
-  const float4 max = (float4)(1.0f, 1.0f, 1.0f, 1.0f);
+  const float4 min_clip = (float4)(0.0f, -exposure_scale, -exposure_scale, 0.0f);
+  const float4 max_clip = (float4)(exposure_scale, exposure_scale, exposure_scale, 1.0f);
+  const float4 min = bounded ? min_clip : (float4)(0.0f, -INFINITY, -INFINITY, INFINITY);
+  const float4 max = bounded ? max_clip : (float4)(INFINITY, INFINITY, INFINITY, 1.0f);
+
   const float4 lmin = (float4)(0.0f, 0.0f, 0.0f, 1.0f);
   const float4 lmax = (float4)(1.0f, 2.0f, 2.0f, 1.0f);       /* max + fabs(min) */
   const float4 halfmax = (float4)(0.5f, 1.0f, 1.0f, 0.5f);    /* lmax / 2.0f */
   const float4 doublemax = (float4)(2.0f, 4.0f, 4.0f, 2.0f);  /* lmax * 2.0f */
   const float opacity2 = opacity*opacity;
 
-  float4 la = clamp(a + fabs(min), lmin, lmax);
-  float4 lb = clamp(b + fabs(min), lmin, lmax);
+  float4 la = clamp((a + fabs(min_clip)) / exposure_scale, lmin, lmax);
```

Comment:
you can't scale Lab vectors like that, a and b represent the chromaticity, they are not linearly connected to L.

Reply:
You're right. Lab vectors are not scaled anymore (the algorithms stay the same as in master). A separate set of not clamping blending modes have been added for scene referred blending.